controllers/machinedeletionremediation_controller.go

```diff
@@ -70,31 +70,32 @@ type MachineDeletionRemediationReconciler struct {
 func (r *MachineDeletionRemediationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	log := r.Log.WithValues("machinedeletionremediation", req.NamespacedName)
 
-	log.Info("reconciling...")
-
-	//fetch the remediation
+	var err error
 	var remediation *v1alpha1.MachineDeletionRemediation
-	if remediation = r.getRemediation(ctx, req); remediation == nil {
-		return ctrl.Result{}, nil
+	if remediation, err = r.getRemediation(ctx, req); remediation == nil || err != nil {
+		return ctrl.Result{}, err
 	}
 
-	var machine *unstructured.Unstructured
-	//Health check was done by NHC
-	if node, err := r.getNodeFromMdr(remediation); err == nil {
-		if machine, err = r.buildMachineFromNode(node); err != nil {
-			r.Log.Error(err, "failed to fetch machine of node", "node name", node.Name)
-			return ctrl.Result{}, err
-		}
-		if !hasControllerOwner(machine) {
-			r.Log.Info("ignoring remediation of machine associated to node, since the machine has no controller owner", "node name", remediation.Name)
-			return ctrl.Result{}, nil
-		}
-
-	} else { //Failed fetching the node
+	var node *v1.Node
+	if node, err = r.getNodeFromMdr(remediation); err != nil {
 		r.Log.Error(err, "failed to fetch node", "node name", remediation.Name)
 		return ctrl.Result{}, err
 	}
+	log.Info("reconciling...")
```

Comment:
NIT: I have no strong preference on moving it to here from above, but maybe update the information to have more meaning? "Successfully fetched MDR CR and it's node" or anything clearer that clarify what happened up to this point in the reconcile loop rather than just decalring that a reconcile has begun which is a better suited argument when this log was in the beginning of the code.

Reply:
Nit: I think it should be at the top

Reply:
The rationale to move it down was that, when at the top, we are not really sure we're going to "reconcile" anything, since it might just be that CR was just deleted.
I agree that a bit more information would be beneficial

Reply:
IMHO we always reconcile the MDR, starting on top of this method. Cases were we don't have to or can do anything are just valid cases to stop reconcile early and should be logged as just that.

Reply:
I move the log back at the top, and added more info when the CR and Node are found